cmake/CMakeLists.txt

```diff
@@ -842,7 +842,7 @@ include(eigen)
 #onnxruntime_EXTERNAL_LIBRARIES could contain onnx, onnx_proto,libprotobuf, cuda/cudnn,
 # dnnl/mklml, onnxruntime_codegen_tvm, tvm, nnvm_compiler and pthread
 # pthread is always at the last
-set(onnxruntime_EXTERNAL_LIBRARIES onnx onnx_proto ${PROTOBUF_LIB} re2::re2)
+set(onnxruntime_EXTERNAL_LIBRARIES -Wl,--no-as-needed onnx onnx_proto ${PROTOBUF_LIB} -Wl,--as-needed re2::re2)
```

Comment:
"--no-as-needed" is not needed

Reply:
Yes. This linker option seem to be causing issue with MVC. I have replaced --no-as-needed with --whole-archive and removed duplicate functions definitions from core/framework/utils.cc. Because /whole_archive options is supported on MSVC, it should work.